controllers/remote/cluster_cache.go

```diff
@@ -396,15 +396,15 @@ func (h *healthCheckInput) validate() {
 // healthCheckCluster will poll the cluster's API at the path given and, if there are
 // `unhealthyThreshold` consecutive failures, will deem the cluster unhealthy.
 // Once the cluster is deemed unhealthy, the cluster's cache is stopped and removed.
-func (m *ClusterCacheTracker) healthCheckCluster(in *healthCheckInput) {
+func (m *ClusterCacheTracker) healthCheckCluster(ctx context.Context, in *healthCheckInput) {
 	// populate optional params for healthCheckInput
 	in.validate()
 
 	unhealthyCount := 0
 
 	runHealthCheckWithThreshold := func() (bool, error) {
 		cluster := &clusterv1.Cluster{}
-		if err := m.client.Get(context.TODO(), in.cluster, cluster); err != nil {
+		if err := m.client.Get(context.Background(), in.cluster, cluster); err != nil {
```

Comment:
`ctx` is available in this function.

```suggestion
		if err := m.client.Get(ctx, in.cluster, cluster); err != nil {
```